tests/assembleCoaddTestUtils.py

```diff
@@ -91,36 +73,18 @@ def get(self, bbox=None, component=None, parameters=None):
             Either the exposure or its metadata, depending on the datasetType.
         """
         if component == 'psf':
-            return self.exposure.getPsf()
+            return self.inMemoryDataset.getPsf()
         elif component == 'visitInfo':
-            return self.exposure.getInfo().getVisitInfo()
+            return self.inMemoryDataset.getInfo().getVisitInfo()
+        bbox = None
         if parameters is not None:
-            if "bbox" in parameters:
-                bbox = parameters["bbox"]
-        exp = self.exposure.clone()
+            bbox = parameters.get("bbox")
+        exp = self.inMemoryDataset.clone()
```

Comment:
Is this override necessary (rather than simply using the straight `InMemoryDatasetHandle`) because of this clone?  I'm wondering if we could avoid this subclass and do the clone below in `makeDataRefList`? https://github.com/lsst/pipe_tasks/blob/6a5135d49b1060558619b0b02ca348b6040e185a/tests/assembleCoaddTestUtils.py#L451

Reply:
Changing that line to:
```python
dataRef = MockWarpReference(exposure.clone(), dataId=dataId)
```
does not fix the problem. It's almost like assembleCoaddTask is modifying the data it retrieves and so subsequent calls are retrieving something different each time.

Reply:
There are three places inside `assembleCoaddTask` that are assuming that each time `get()` is called it is getting a brand new exposure that it can modify. This suggests that we are required to keep a special subclass of InMemoryDatasetHandle for now until there is some way that we can indicate that a clone/copy is needed. We could have a new parameter to the InMemoryDatasetHandle constructor that indicates whether a copy/clone is required or not. This would then have to be something that delegate has to understand since I don't think we can assume that `copy.deepcopy` would always work (a parameter to `get()` seems painful since DeferredDatasetHandle for file datastore will always give you a new copy -- it can't do anything else)

Reply:
I was just thinking the same thing about InMemoryDatasetHandle.

Reply:
Will not solve this on the current ticket.